internal/services/azurestackhci/stack_hci_deployment_setting_resource.go

```diff
@@ -0,0 +1,1365 @@
+						Default:  true,
+						ForceNew: true,
+					},
+
+					"credential_guard_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  false,
+						ForceNew: true,
+					},
+
+					"drift_control_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  true,
+						ForceNew: true,
+					},
+
+					"drtm_protection_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  true,
+						ForceNew: true,
+					},
+
+					"hvci_protection_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  true,
+						ForceNew: true,
+					},
+
+					"side_channel_mitigation_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  true,
+						ForceNew: true,
+					},
+
+					"smb_signing_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  true,
+						ForceNew: true,
+					},
+
+					"smb_cluster_encryption_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  false,
+						ForceNew: true,
+					},
+
+					"wdac_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  true,
+						ForceNew: true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (StackHCIDeploymentSettingResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r StackHCIDeploymentSettingResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 6 * time.Hour,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AzureStackHCI.DeploymentSettings
+
+			var config StackHCIDeploymentSettingModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			stackHCIClusterId, err := deploymentsettings.ParseClusterID(config.StackHCIClusterId)
+			if err != nil {
+				return err
+			}
+			id := deploymentsettings.NewDeploymentSettingID(stackHCIClusterId.SubscriptionId, stackHCIClusterId.ResourceGroupName, stackHCIClusterId.ClusterName, "default")
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			payload := deploymentsettings.DeploymentSetting{
+				Properties: &deploymentsettings.DeploymentSettingsProperties{
+					ArcNodeResourceIds: config.ArcResourceIds,
+					DeploymentMode:     deploymentsettings.DeploymentModeValidate,
+					DeploymentConfiguration: deploymentsettings.DeploymentConfiguration{
+						Version:    pointer.To(config.Version),
+						ScaleUnits: expandDeploymentSettingScaleUnits(config.ScaleUnit),
+					},
+				},
+			}
+
+			// the resource may exist even validation error
+			metadata.SetID(id)
```

Comment:
i'm not sure what this means, we should not be setting the ID until after the resource is created?

Reply:
updated, SetID is moved after resource is created.